presto-hive/src/test/java/com/facebook/presto/hive/TestHiveIntegrationSmokeTest.java

```diff
@@ -2769,6 +2770,76 @@ public void testGroupedExecution()
         }
     }
 
+    @Test
+    public void testWindowFunction()
```

Comment:
Would it be possible to add another test that would be independent from Hive connector? One that is aimed at testing engine only?

Your original issue was about bucketed tables (hive specific), but from what i understand, the problem is more generic, and should be applicable to case like window-after-aggregation or window-after-partitioned-join.

Reply:
That sounds fair. I'll add tests in `TestLogicalPlanner`.

Reply:
I don't see why JOIN (alone) could be partitioned on stream level. 
The build side is being hash-partitioned locally for the sake of building the lookup source, but the probe side is either source distributed (if this is broadcast join directly above table scan and spill is disabled) or round-robin. In neither case we should claim steam level partitioning. Am i missing something?

@sopel39 ?

Reply:
I'm using this query for testing: `SELECT rank() OVER (partition by o.orderkey) FROM orders o JOIN lineitem l ON o.orderkey = l.orderkey`. 

From debugger, I could see that in `ActualProperties`, `global.nodePartitioning` and `global.streamPartitioning` are the same (`["orderkey"]`), which holds since `TableScanNode`'s `PlanWithProperties` is created. I guess I missed some obvious pieces. Maybe I need some way to make the probe side's input not stream partitioned on the desired column?

Reply:
Maybe tpch generates data partitioned over `oderkey`? Did you try joining on a different column (the join condition doesn't have to be sensible, it can be pretty anything)

Reply:
You are right that the data seems to be partitioned over `orderkey`. I tried `SELECT rank() OVER (partition by o.orderstatus) FROM orders o JOIN lineitem l ON o.orderstatus = l.linestatus` as well, and looks like the global properties now changed to stream partitioned after partitioned join analysis in `AddExchanges`:

https://github.com/prestodb/presto/blob/7dfbe1be3c61455047827335c5dadd94125e0c48/presto-main/src/main/java/com/facebook/presto/sql/planner/optimizations/AddExchanges.java#L687-L689

Distributed plan is the same for before/after:
```
...
 Fragment 1 [HASH]                                                                                                                                                                  
     Output layout: [rank]                                                                                                                                                          
     Output partitioning: SINGLE []                                                                                                                                                 
     Grouped Execution: false                                                                                                                                                       
     - Project[] => [rank:bigint]                                                                                                                                                   
         - Window[partition by (orderstatus)][$hashvalue] => [orderstatus:varchar(1), $hashvalue:bigint, rank:bigint]                                                               
                 rank := rank() RANGE UNBOUNDED_PRECEDING CURRENT_ROW                                                                                                               
             - LocalExchange[HASH][$hashvalue] ("orderstatus") => orderstatus:varchar(1), $hashvalue:bigint                                                                         
                 - InnerJoin[("orderstatus" = "linestatus")][$hashvalue_9, $hashvalue_11] => [orderstatus:varchar(1), $hashvalue_9:bigint]                                          
                         Distribution: PARTITIONED                                                                                                                                  
                     - RemoteSource[2] => [orderstatus:varchar(1), $hashvalue_9:bigint]                                                                                             
                     - LocalExchange[HASH][$hashvalue_11] ("linestatus") => linestatus:varchar(1), $hashvalue_11:bigint                                                             
                         - RemoteSource[3] => [linestatus:varchar(1), $hashvalue_12:bigint]
...
```